magenta-lib/src/main/scala/magenta/deployment_type/CloudFormation.scala

```diff
@@ -69,18 +75,28 @@ object CloudFormation extends DeploymentType {
       implicit val artifactClient = resources.artifactClient
       val reporter = resources.reporter
 
-      val stackName = target.stack.nameOption.filter(_ => prependStackToCloudFormationStackName(pkg, target, reporter))
-      val stageName = Some(target.parameters.stage.name).filter(_ => appendStageToCloudFormationStackName(pkg, target, reporter))
-      val cloudFormationStackNameParts = Seq(stackName, Some(cloudFormationStackName(pkg, target, reporter)), stageName).flatten
-      val fullCloudFormationStackName = cloudFormationStackNameParts.mkString("-")
+      val cloudFormationStackLookupStrategy = {
+        if (cloudformationStackByTags(pkg, target, reporter)) {
+          LookupByTags(pkg, target, reporter)
+        } else {
+          LookupByName(
+            target.stack,
```

Comment:
I was just making intelliJ's default scala analysis happy - it likes boolean params to be named. I think the others are pretty obvious.